src/main/scala/mesosphere/marathon/event/Events.scala

```diff
@@ -40,6 +40,14 @@ object EventModule {
   final val busName = "events"
 }
 
+/** Local leadership events. They are not delivered via the event endpoints. */
+sealed trait LeadershipStateEvent
+
+object LeadershipStateEvent {
+  case object ElectedAsLeader
```

Comment:
I thought about that but I think it is something different that might be useful on its own.

These are events from the viewpoint of the Marathon instance processing these. They are not global events. You could turn them into global events but it gets more complicated.